### PR TITLE
Move all contribution info and dev guides to `CONTRIBUTING.rst`

### LINKED ISSUE #1039: Add `CONTRIBUTING.md`
I think we should move Contributing section from `README.md` to `CONTRIBUTING.md` and add some local development instructions (how to fork this project, how to create a venv, how to create a pull request and so on)

### PR DESCRIPTION
**Pull request by hirotasoshu**
Resolves #1039

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -77,30 +77,7 @@ If you have questions or have trouble using the app please file a bug report at:
 https://github.com/jazzband/django-axes/issues
 
 
-Contributions
+Contributing
 -------------
```
Comment (aleksihakli) [inline README.rst:81]: The divider line is too short and rST bakery might fail, could you adjust it as well?

**Comment by aleksihakli** [comment]
LGTM! I left one comment, if you have the chance of adjusting the header styling that'd be great.

File: README.rst (hunk 1 above)
Comment (hirotasoshu) [reply README.rst]: Fixed)

**Comment by aleksihakli** [comment]
Thank you, upstreaming 👍

**Pull request merged**